src/basic/fd-util.c

```diff
@@ -282,6 +282,42 @@ static int close_all_fds_frugal(const int except[], size_t n_except) {
         return r;
 }
 
+int close_all_fds_by_proc(const int except[], size_t n_except) {
```

Comment:
So, hmm, if you're bumping the kernel baseline for 259 anyways in https://github.com/systemd/systemd/pull/38616/commits/15701f1cdb644e681ecd17dcc64575a421858065, the whole fallback logic can be killed for good?